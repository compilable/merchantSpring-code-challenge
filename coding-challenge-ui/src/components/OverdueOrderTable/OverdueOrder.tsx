export type OverdueOrder = {
    id: string;
    marketPlace: string,
    country: string,
    shopName: string,
    orderId: string,
    orderValue: string,
    items: number,
    destination: string;
    daysOverdue: number;
};