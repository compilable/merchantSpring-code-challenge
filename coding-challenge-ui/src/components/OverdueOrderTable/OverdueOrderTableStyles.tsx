import { forwardRef } from "react";
import { Icons } from "@material-table/core";
import { ArrowDownward } from "@material-ui/icons";

const OverdueOrderTableStyles = () => {

    const tableOptions = () => {
        return (
            {
                paging: true,
                pageSize: 5,
                emptyRowsWhenPaging: false,
                pageSizeOptions: [5],
                showTitle: true,
                search: false,
                toolbar: true,
                align: 'center',
                headerStyle: {
                    backgroundColor: '#cccccc',
                    color: 'rgb(77 71 71)'
                }
            });
    }

    const tableIcons: Icons = {
        SortArrow: forwardRef((props, ref) => <ArrowDownward {...props} ref={ref} />),
    };

    const tableStyles = {
        margin: "40px auto",
    }

    return { tableOptions, tableIcons,tableStyles }
}

export default OverdueOrderTableStyles



