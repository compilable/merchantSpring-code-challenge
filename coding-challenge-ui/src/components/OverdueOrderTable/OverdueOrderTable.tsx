import React from "react";
import MaterialTable, { Column } from "@material-table/core";
import { Container } from "@material-ui/core";
import {OverdueOrder} from "./OverdueOrder"

import OverdueOrderTableStyles from "./OverdueOrderTableStyles"

const columns: Array<Column<OverdueOrder>> = [
  {
    title: 'MARKETPLACE', align:'justify', field: 'marketPlace', render: rowData => {
      return(<>
        <img alt={rowData.country} src={`https://flagcdn.com/w20/${rowData.country}.png`} style={{ width: 20 }} /> <span>{rowData.marketPlace}</span>
      </>)
    }
  },
  { title: "STORE",align:'justify', field: "shopName" },
  { title: "ORDER ID",align:'justify', field: "orderId" },
  { title: "ORDER VALUE",align:'justify', field: "orderValue" , type:"currency" },
  { title: "ITEMS", align:'justify',field: "items" },
  { title: "DESTINATION", align:'justify',field: "destination" },
  { title: "DAYS OVERDUE", align:'center',field: "daysOverdue", render: rowData => (<span style={{color:'red'}}>{rowData.daysOverdue}</span>) }
];

interface Props{
  heading:string,
  orders: Array<OverdueOrder>
}

const OverdueOrderTable: React.FC<Props> = ({orders,heading})=> {
  const {tableIcons, tableOptions,tableStyles}= OverdueOrderTableStyles()
  return (
    <Container style={tableStyles}>
      <MaterialTable
        title={heading}
        columns={columns}
        data={orders}
        icons={tableIcons}
        options={tableOptions()} />
    </Container>
  );
}

export default OverdueOrderTable