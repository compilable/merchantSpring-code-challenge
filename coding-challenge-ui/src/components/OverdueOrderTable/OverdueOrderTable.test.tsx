import OverdueOrderTable from './OverdueOrderTable';
import { render } from '@testing-library/react';
import toJSON from "enzyme-to-json"
import faker from "faker"
import { shallow } from 'enzyme';
faker.seed(0)

let overdueOrderData:any

describe('test Overdue Orders table loaded with data', () => {
  overdueOrderData = ([...new Array(10)].map(() => ({
    id: faker.datatype.uuid(),
    marketPlace: `https://flagcdn.com/w20/${faker.address.countryCode().toLowerCase()}.png`,
    country: faker.address.countryCode(),
    shopName:faker.company.companyName(),
    orderId: faker.vehicle.vrm(),
    orderValue:faker.commerce.price(),
    items:Math.floor((Math.random() * 50) + 1),
    destination: faker.address.cityName(),
    daysOverdue: (Math.floor((Math.random() * 10) + 1))
  })))

  let tableWithData:any
  beforeEach(() => {
    tableWithData  = render(<OverdueOrderTable heading="Overdue Orders" orders={overdueOrderData} />);

    console.debug(tableWithData)
  });
  it('table has loaded', () => {
    expect(tableWithData.getAllByText("Overdue Orders"))
  });
});

test("has the table data loaded", () => {
  overdueOrderData = [
    {
      "country": "LI",
      "daysOverdue": 5,
      "destination": "Carmichael",
      "id": "a73cf3be-453f-4870-b326-b5a73429cdb1",
      "items": 4,
      "marketPlace": "https://flagcdn.com/w20/nr.png",
      "orderId": "SS34OVE",
      "orderValue": "205.00",
      "shopName": "Cole Inc",
    },
    {
      "country": "SS",
      "daysOverdue": 2,
      "destination": "Summerville",
      "id": "2715bf0c-bb1e-431b-8b90-f3443a1108e0",
      "items": 42,
      "marketPlace": "https://flagcdn.com/w20/no.png",
      "orderId": "PE05MPZ",
      "orderValue": "788.00",
      "shopName": "Wyman, Grady and Prosacco",
    }
  ]
  const wrapper = shallow(<OverdueOrderTable heading="Overdue Orders" orders={overdueOrderData} />);
  expect(toJSON(wrapper)).toMatchSnapshot()
});