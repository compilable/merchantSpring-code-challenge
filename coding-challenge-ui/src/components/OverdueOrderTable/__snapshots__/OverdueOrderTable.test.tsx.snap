// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has the table component loaded 1`] = `
<WithStyles(ForwardRef(Container))
  style={
    Object {
      "margin": "40px auto",
    }
  }
>
  <WithStyles(Component)
    columns={
      Array [
        Object {
          "align": "justify",
          "field": "marketPlace",
          "render": [Function],
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 0,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 0,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "MARKETPLACE",
        },
        Object {
          "align": "justify",
          "field": "shopName",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 1,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 1,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "STORE",
        },
        Object {
          "align": "justify",
          "field": "orderId",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 2,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 2,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ORDER ID",
        },
        Object {
          "align": "justify",
          "field": "orderValue",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 3,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 3,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ORDER VALUE",
          "type": "currency",
        },
        Object {
          "align": "justify",
          "field": "items",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 4,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 4,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ITEMS",
        },
        Object {
          "align": "justify",
          "field": "destination",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 5,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 5,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "DESTINATION",
        },
        Object {
          "align": "center",
          "field": "daysOverdue",
          "render": [Function],
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 6,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 6,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "DAYS OVERDUE",
        },
      ]
    }
    data={
      Array [
        Object {
          "country": "LI",
          "daysOverdue": 5,
          "destination": "Carmichael",
          "id": "a73cf3be-453f-4870-b326-b5a73429cdb1",
          "items": 4,
          "marketPlace": "https://flagcdn.com/w20/nr.png",
          "orderId": "SS34OVE",
          "orderValue": "205.00",
          "shopName": "Cole Inc",
        },
        Object {
          "country": "SS",
          "daysOverdue": 2,
          "destination": "Summerville",
          "id": "2715bf0c-bb1e-431b-8b90-f3443a1108e0",
          "items": 42,
          "marketPlace": "https://flagcdn.com/w20/no.png",
          "orderId": "PE05MPZ",
          "orderValue": "788.00",
          "shopName": "Wyman, Grady and Prosacco",
        },
      ]
    }
    icons={
      Object {
        "SortArrow": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    options={
      Object {
        "align": "center",
        "emptyRowsWhenPaging": false,
        "headerStyle": Object {
          "backgroundColor": "#cccccc",
          "color": "rgb(77 71 71)",
        },
        "pageSize": 5,
        "pageSizeOptions": Array [
          5,
        ],
        "paging": true,
        "search": false,
        "showTitle": true,
        "toolbar": true,
      }
    }
    title="Overdue Orders"
  />
</WithStyles(ForwardRef(Container))>
`;

exports[`has the table data loaded 1`] = `
<WithStyles(ForwardRef(Container))
  style={
    Object {
      "margin": "40px auto",
    }
  }
>
  <WithStyles(Component)
    columns={
      Array [
        Object {
          "align": "justify",
          "field": "marketPlace",
          "render": [Function],
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 0,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 0,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "MARKETPLACE",
        },
        Object {
          "align": "justify",
          "field": "shopName",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 1,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 1,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "STORE",
        },
        Object {
          "align": "justify",
          "field": "orderId",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 2,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 2,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ORDER ID",
        },
        Object {
          "align": "justify",
          "field": "orderValue",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 3,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 3,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ORDER VALUE",
          "type": "currency",
        },
        Object {
          "align": "justify",
          "field": "items",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 4,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 4,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "ITEMS",
        },
        Object {
          "align": "justify",
          "field": "destination",
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 5,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 5,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "DESTINATION",
        },
        Object {
          "align": "center",
          "field": "daysOverdue",
          "render": [Function],
          "tableData": Object {
            "additionalWidth": 0,
            "columnOrder": 6,
            "filterValue": undefined,
            "groupOrder": undefined,
            "groupSort": "asc",
            "id": 6,
            "initialWidth": "calc((100% - 0px) / 7)",
            "width": "calc((100% - 0px) / 7)",
            "widthPx": NaN,
          },
          "title": "DAYS OVERDUE",
        },
      ]
    }
    data={
      Array [
        Object {
          "country": "LI",
          "daysOverdue": 5,
          "destination": "Carmichael",
          "id": "a73cf3be-453f-4870-b326-b5a73429cdb1",
          "items": 4,
          "marketPlace": "https://flagcdn.com/w20/nr.png",
          "orderId": "SS34OVE",
          "orderValue": "205.00",
          "shopName": "Cole Inc",
        },
        Object {
          "country": "SS",
          "daysOverdue": 2,
          "destination": "Summerville",
          "id": "2715bf0c-bb1e-431b-8b90-f3443a1108e0",
          "items": 42,
          "marketPlace": "https://flagcdn.com/w20/no.png",
          "orderId": "PE05MPZ",
          "orderValue": "788.00",
          "shopName": "Wyman, Grady and Prosacco",
        },
      ]
    }
    icons={
      Object {
        "SortArrow": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    options={
      Object {
        "align": "center",
        "emptyRowsWhenPaging": false,
        "headerStyle": Object {
          "backgroundColor": "#cccccc",
          "color": "rgb(77 71 71)",
        },
        "pageSize": 5,
        "pageSizeOptions": Array [
          5,
        ],
        "paging": true,
        "search": false,
        "showTitle": true,
        "toolbar": true,
      }
    }
    title="Overdue Orders"
  />
</WithStyles(ForwardRef(Container))>
`;
